### PR TITLE
Enhancement and Test Suite for ResourceTable

### PR DESCRIPTION
**Pull request by VaibhavMalik4187**
## PR Type:
Tests

___
## PR Description:
This PR introduces enhancements and a comprehensive test suite for the `resourcetable` file. The key changes are:
- Enhancements to the `generateResourceHeader` function for dynamic header length, efficient header creation, and clearer code structure.
- Enhancements to the `appendFailedPathsIfNotInPaths` function to create a set for efficient checking of existing paths and appending failed paths if they are not already present.
- Addition of a comprehensive test suite for the `resourcetable` file, covering various functions and scenarios.

___
## PR Main Files Walkthrough:
<details> <summary>files:</summary>

- `core/pkg/resultshandling/printer/v2/resourcetable.go`: The `generateResourceHeader` function has been enhanced for dynamic header length, efficient header creation, and clearer code structure. The `appendFailedPathsIfNotInPaths` function has been enhanced to create a set for efficient checking of existing paths and appending failed paths if they are not already present.
- `core/pkg/resultshandling/printer/v2/resourcetable_test.go`: A comprehensive test suite has been added for the `resourcetable` file. The tests cover various functions such as `appendFailedPathsIfNotInPaths`, `AssistedRemediationPathsToString`, `reviewPathsToString`, `deletePathsToString`, `fixPathsToString`, `failedPathsToString`, `shortFormatResource`, `generateResourceHeader`, and `generateResourceRows_Loop`. The tests cover various scenarios and edge cases to ensure the robustness of the code.
</details>

### CODE REVIEW

File: core/pkg/resultshandling/printer/v2/resourcetable.go [hunk 1]
```diff
@@ -112,17 +111,14 @@ func generateResourceRows(controls []resourcesresults.ResourceAssociatedControl,
 }
 
 func generateResourceHeader(short bool) []string {
-	var headers []string
+	headers := make([]string, 0)
```
Comment (matthyx) [inline core/pkg/resultshandling/printer/v2/resourcetable.go:114]: you could also just declare with a var...

File: core/pkg/resultshandling/printer/v2/resourcetable.go [hunk 2]
```diff
@@ -212,10 +208,18 @@ func AssistedRemediationPathsToString(control *resourcesresults.ResourceAssociat
...
+	// Create a set to efficiently check if a failed path already exists in the paths slice
+	pathSet := make(map[string]struct{})
+	for _, path := range paths {
+		pathSet[path] = struct{}{}
+	}
+
+	// Append failed paths if they are not already present
 	for _, failedPath := range failedPaths {
-		if !slices.Contains(paths, failedPath) {
+		if _, ok := pathSet[failedPath]; !ok {
```
Comment (matthyx) [inline core/pkg/resultshandling/printer/v2/resourcetable.go:219]: nice

File: core/pkg/resultshandling/printer/v2/resourcetable_test.go [hunk 3]
```diff
@@ -0,0 +1,485 @@
...
+			},
+			expectedLen: 1,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			rows := generateResourceRows(tt.controls, &tt.summaryDetails)
+			assert.Equal(t, tt.expectedLen, len(rows))
+			// assert.Equal(t, updatedPaths, testcase.expectedPaths)
```
Comment (matthyx) [inline core/pkg/resultshandling/printer/v2/resourcetable_test.go:482]: why is that commented out?

**Review by matthyx — CHANGES_REQUESTED** [review]

File: core/pkg/resultshandling/printer/v2/resourcetable.go (hunk 1 above)
Comment (VaibhavMalik4187) [reply core/pkg/resultshandling/printer/v2/resourcetable.go:114]: Agreed, it is just that make is slightly more performant than var declaration.

File: core/pkg/resultshandling/printer/v2/resourcetable_test.go (hunk 3 above)
Comment (VaibhavMalik4187) [reply core/pkg/resultshandling/printer/v2/resourcetable_test.go]: I forgot to remove it, thanks for pointing this out.

**Review by matthyx — APPROVED** [review]

**Pull request merged**